app/celery/provider_tasks.py

```diff
@@ -51,6 +52,9 @@ def deliver_sms(self, notification_id, sms_sender_id=None):
         )
         notification = notifications_dao.get_notification_by_id(notification_id)
         check_and_queue_callback_task(notification)
+    except NullValueForNonConditionalPlaceholderException:
+        msg = handle_non_retryable(notification_id, 'deliver_email')
```

Comment:
```suggestion
        msg = handle_non_retryable(notification_id, 'deliver_sms')
```
copy-paste issue, pass correct method name